app/repositories/scores.py

```diff
@@ -182,26 +237,28 @@ async def fetch_many(
         params["page_size"] = page_size
         params["offset"] = (page - 1) * page_size
 
-    recs = await app.state.services.database.fetch_all(query, params)
-    return [dict(rec) for rec in recs]
+    scores = await app.state.services.database.fetch_all(query, params)
+    return cast(list[Score], scores) if scores is not None else None
 
 
 async def update(
     id: int,
-    pp: float | None = None,
-    status: int | None = None,
-) -> dict[str, Any] | None:
-    query = """\
+    pp: float | _UnsetSentinel = UNSET,
+    status: int | _UnsetSentinel = UNSET,
+) -> Score | None:
+    """Update an existing score."""
+    update_fields = ScoreUpdateFields = {}
+    if not isinstance(pp, _UnsetSentinel):
+        update_fields["pp"] = pp
+    if not isinstance(status, _UnsetSentinel):
+        update_fields["status"] = status
+
+    query = f"""\
         UPDATE scores
-           SET pp = COALESCE(:pp, pp),
-               status = COALESCE(:status, status)
+           SET {",".join(f"{k} = COALESCE(:{k}, {k})" for k in update_fields)}
          WHERE id = :id
     """
-    params = {
-        "id": id,
-        "pp": pp,
-        "status": status,
-    }
+    values = {"id": id} | update_fields
     await app.state.services.database.execute(query, params)
```

Comment:
in this case `params` was renamed to `values` which causes 

![image](https://i.cmyui.xyz/RnxshQbKEro.png)

you can get these errors to show up in your vs code by setting the `type checking mode` setting to `"basic"`

![image](https://i.cmyui.xyz/NlvJI-8YZ0uHEg.png)